deploy/report/api/main.py

```diff
@@ -0,0 +1,307 @@
+
+# This endpoint serves the UI
+@app.get("/", response_class=HTMLResponse)
+async def get_home() -> HTMLResponse:
+    """Return home page for cyclops model report app.
+
+    Returns
+    -------
+    HTMLResponse
+        Home page for cyclops model report app.
+
+    """
+    return templates.TemplateResponse(
+        "test_report.html", {"request": {"method": "POST"}}
+    )
+
+
+@app.post("/evaluate")
+async def evaluate_result(data: EvaluationInput) -> None:
+    """Calculate metric and return result from request body.
+
+    Parameters
+    ----------
+    data : EvaluationInput
+        Input data for evaluation.
+
+    Raises
+    ------
+    HTTPException
+        If there is an internal server error.
+
+    """
+    try:
+        # Create a dictionary with all data
+        df_dict = {
+            "target": data.target,
+            "preds_prob": data.preds_prob,
+            **data.metadata,
+        }
+
+        # Create DataFrame
+        df = pd.DataFrame(df_dict)
+
+        _eval(df)
+        LOGGER.info("Generated report.")
+    except Exception as e:
+        LOGGER.error(f"Error during evaluation: {str(e)}")
+        raise HTTPException(status_code=500, detail="Internal server error") from e
+
+
+@app.get("/evaluate")
+async def get_report() -> HTMLResponse:
+    """Return latest updated model report.
+
+    Returns
+    -------
+    HTMLResponse
+        Latest updated model report as HTML response.
+
+    """
+    return templates.TemplateResponse(
+        "test_report.html", {"request": {"method": "GET"}}
+    )
+
+
+def _export(report: ModelCardReport) -> None:
+    """Prepare and export report file."""
+    if not os.path.exists("./cyclops_report"):
+        LOGGER.info("Creating report for the first time!")
+    report_path = report.export(
+        output_filename="test_report.html",
+        synthetic_timestamp=str(datetime.today()),
+        last_n_evals=3,
+    )
+    shutil.copy(f"{report_path}", TEMPLATES_PATH)
+
+
+def _eval(df: pd.DataFrame) -> None:
+    """Evaluate and return report.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        DataFrame containing target, preds_prob and metadata columns.
+
+    """
+    report = ModelCardReport()
+    data = Dataset.from_pandas(df)
+    metric_names = [
+        "binary_accuracy",
+        "binary_precision",
+        "binary_recall",
+        "binary_f1_score",
+    ]
+    metrics = [
+        create_metric(metric_name, experimental=True) for metric_name in metric_names
+    ]
+    metric_collection = MetricDict(metrics)
+    spec_list = [
+        {
+            "Age": {
+                "min_value": 30,
+                "max_value": 50,
+                "min_inclusive": True,
+                "max_inclusive": False,
+            },
+        },
+        {
+            "Age": {
+                "min_value": 50,
+                "max_value": 70,
+                "min_inclusive": True,
+                "max_inclusive": False,
+            },
+        },
```

Comment:
Would be cool if the metric names and spec list could be provided by the user. Probably not for this PR though.

Reply:
Yes, it's going to be added in later PRs.